### PR TITLE
Updated every reference from cover to poster where applicable

### LINKED ISSUE #6370: Change "Cover Image" to "Poster image" in the Document Panel
## Bug Description

Our current terminology of "Cover Image" doesn't match the overall documentation in Web Stories that use "Poster image" for this field. We should change the UI to say Poster image. (noticed the casing)
image

Also, make "Required" have capital R. 

## Additional Context

- Plugin Version:
- WordPress Version:
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by barklund**
## Context

To avoid confusion, we're renaming "Cover image" to "Poster image" - and changing every reference to "story cover image" to simply be "poster image".

Also fixing a few typos and other stuff while at it (mostly in tests).

## User-facing changes

Input text has changed next to the image selector as well as messaging in pre-publish panel:

| | Before | After |
|-|-|-|
| Input | Screen Shot 2021-02-12 at 14 35 38 | Screen Shot 2021-02-12 at 15 12 00 |
| Missing image | Screen Shot 2021-02-12 at 14 35 43 | Screen Shot 2021-02-12 at 14 35 15 |
| Wrong size | Screen Shot 2021-02-12 at 14 35 56 | Screen Shot 2021-02-12 at 14 36 29 |

### QA

This PR can be tested by following these steps:

1. Create new story
2. Go to document panel
3. Observe label next to poster image input
4. Go to prepublish panel
5. Observe message around missing poster image

### UAT

- [x] UAT should use the same steps as above.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6370

### CODE REVIEW

**Review by BrittanyIRL — APPROVED** [review]
LGTM, changes are as described  🚀

File: assets/src/edit-story/app/prepublish/constants.js [hunk 1]
```diff
@@ -24,8 +24,8 @@ export const MIN_STORY_PAGES = 4;
 export const MAX_STORY_PAGES = 30;
 export const MAX_STORY_TITLE_LENGTH_WORDS = 10;
 export const MAX_STORY_TITLE_LENGTH_CHARS = 40;
-const COVER_DIMENSION_WIDTH_PX = 640;
-const COVER_DIMENSION_HEIGHT_PX = 853;
+const POSTER_DIMENSION_WIDTH_PX = 640;
+const POSTER_DIMENSION_HEIGHT_PX = 853;
```
Comment (samwhale) [inline assets/src/edit-story/app/prepublish/constants.js:28]: nice, thanks for updating the variable names as well

File: assets/src/edit-story/app/prepublish/error/test/metadata.js [hunk 2]
```diff
@@ -20,21 +20,21 @@
...
+    expect(testMissingPoster).not.toBeUndefined();
+    expect(testMissingPoster.message).toMatchInlineSnapshot(
+      `"Add poster image"`
     );
-    expect(testMissingCover.storyId).toStrictEqual(testStory.id);
+    expect(testMissingPoster.storyId).toStrictEqual(testStory.id);
   });
 
-  it("should return an error-type guidance message if the story is missing it's title", () => {
+  it('should return an error-type guidance message if the story is missing its title', () => {
```
Comment (samwhale) [inline assets/src/edit-story/app/prepublish/error/test/metadata.js:37]: ✅

**Review by samwhale — APPROVED** [review]
Very thorough update ✨

**Review by embarks — APPROVED** [review]
doing `ctrl+f` cover on this branch looks good to me 👍 too bad the 3P image API is called `Coverr` 😂

**Pull request merged**